Add ability to reduce the target video's frame rate

niekdev opened the pull request:
This PR adds a simple `FrameDropper` as recommended by alexvaliskov in [issue #120](https://github.com/linkedin/LiTr/issues/120#issuecomment-859318538). This allows users to reduce the target video's frame rate, which has been a requested feature for some time.

I've also included some tests and the ability to alter the frame rate in the demo app, which makes it easy to runtime test this feature.

izzytwosheds commented on litr/src/main/java/com/linkedin/android/litr/transcoder/VideoTrackTranscoder.java:
```diff
@@ -267,4 +297,10 @@ private int writeEncodedOutputFrame() throws TrackTranscoderException {
+
+    @VisibleForTesting
+    @Nullable
+    public FrameDropper getFrameDropper() {
```
nip: can you rename this to `createFrameDropper`?

NiekAkerboom replied:
I think that you meant renaming `initFrameDropper` -> `createFrameDropper`? This here is just a getter visibly only to tests, it'll return the current `frameDropper` (nullable) object.

I've updated the name of `initFrameDropper` to `createFrameDropper`: [VideoTrackTranscoder.java#L87](https://github.com/linkedin/LiTr/blob/a6d524823017dc2cf30ed85026e7d1ad3af4ace9/litr/src/main/java/com/linkedin/android/litr/transcoder/VideoTrackTranscoder.java#L87)

niekdev replied:
I think that you meant renaming `initFrameDropper` -> `createFrameDropper`? This here is just a getter visibly only to tests, it'll return the current `frameDropper` (nullable) object.

I've updated the name of `initFrameDropper` to `createFrameDropper`: [VideoTrackTranscoder.java#L87](https://github.com/linkedin/LiTr/blob/a6d524823017dc2cf30ed85026e7d1ad3af4ace9/litr/src/main/java/com/linkedin/android/litr/transcoder/VideoTrackTranscoder.java#L87)

izzytwosheds commented on litr/src/main/java/com/linkedin/android/litr/render/FrameDropper.kt:
```diff
@@ -0,0 +1,36 @@
+
+    companion object {
+
+        fun getDefaultFrameDropper(
```
Why do we need this static factory method instead of just having a `DefaultFrameDropper` implementation and creating it directly in `TrackTranscoder`?

NiekAkerboom replied:
Good question, don't have a valid answer for that 😅. Updated it at: [FrameDropper.kt#L14-L33](https://github.com/linkedin/LiTr/blob/a6d524823017dc2cf30ed85026e7d1ad3af4ace9/litr/src/main/java/com/linkedin/android/litr/render/FrameDropper.kt#L14-L33)

niekdev replied:
Good question, don't have a valid answer for that 😅. Updated it at: [FrameDropper.kt#L14-L33](https://github.com/linkedin/LiTr/blob/a6d524823017dc2cf30ed85026e7d1ad3af4ace9/litr/src/main/java/com/linkedin/android/litr/render/FrameDropper.kt#L14-L33)

izzytwosheds commented on litr/src/main/java/com/linkedin/android/litr/transcoder/VideoTrackTranscoder.java:
```diff
@@ -40,8 +42,9 @@ public class VideoTrackTranscoder extends TrackTranscoder {
+    @Nullable private FrameDropper frameDropper;
 
-    VideoTrackTranscoder(@NonNull MediaSource mediaSource,
+    public VideoTrackTranscoder(@NonNull MediaSource mediaSource,
```
Does this have to become public?

NiekAkerboom replied:
Nope, apologies, I added my test to the wrong package so it wasn't visible. Updated at: [VideoTrackTranscoder.java#L48](https://github.com/linkedin/LiTr/blob/a6d524823017dc2cf30ed85026e7d1ad3af4ace9/litr/src/main/java/com/linkedin/android/litr/transcoder/VideoTrackTranscoder.java#L48) and I've moved the frame dropper tests to the correct package.

niekdev replied:
Nope, apologies, I added my test to the wrong package so it wasn't visible. Updated at: [VideoTrackTranscoder.java#L48](https://github.com/linkedin/LiTr/blob/a6d524823017dc2cf30ed85026e7d1ad3af4ace9/litr/src/main/java/com/linkedin/android/litr/transcoder/VideoTrackTranscoder.java#L48) and I've moved the frame dropper tests to the correct package.

NiekAkerboom commented:
Hey @izzytwosheds, thank you for your comments. I have addressed your concerns and force pushed the commits to make the git history a bit nicer. I've replied to each comment with a link to the adjusted lines.

niekdev commented:
Hey @izzytwosheds, thank you for your comments. I have addressed your concerns and force pushed the commits to make the git history a bit nicer. I've replied to each comment with a link to the adjusted lines.
